extensions/resume.md

```diff
@@ -0,0 +1,255 @@
+    C2 - C: RESUME A8KgnZPYDaRiGMzZWLu2frVvtN7lbCxO3hTwGLO 2017-04-13T15:12:51.620Z
+    C2 - S: :irc.example.com RESUME ERR :Cannot resume connection, token is not valid
+    C2 - C: AUTHENTICATE PLAIN
+    C2 - S: :irc.example.com AUTHENTICATE +
+    C2 - C: AUTHENTICATE YnVubnkAYnVubnkAYnVubnk=
+    C2 - S: :irc.example.com 900 dan-backup-nick * bunny :You are now logged in as bunny
+    C2 - S: :irc.example.com 903 dan-backup-nick :SASL authentication successful
+    C2 - S: :irc.example.com 001 dan-backup-nick :Welcome to the Internet Relay Network dan-backup-nick
+    ... regular connection ...
+
+
+## Implementation Considerations
+
+This section notes considerations software authors will need to take into account while implementing this specification. This section is non-normative.
+
+Right now, when clients detect that their connection to the server may have dropped they tend to send a `QUIT` command, close their current connection and then create a new connection to the server. In cases where the server supports resuming connections, clients may find it more useful to attempt to establish a new link to the server and resume the connection before closing their old one. If this is done, clients should be able to better take advantage of connection resumption.
+
+In addition, users sometimes manually reconnect when they see that there is lag on their connection. In these cases, clients may also wish to do the above rather than closing the connection and then reconnecting.
+
+When clients see a `RESUMED` message for another client which contains a timestamp, they can calculate how much time has passed since the timestamp and the current time and then display this next to the reconnect notice. Displaying this can assist users in knowing how much message history has been lost in private queries and channels.
+
+A client that disconnects and reconnects to the server should explicitly display the reconnection, even if they're able to resume successfully. This is so that the user knows why they may be missing message history and similar issues.
+
+Servers may wish to check the new hostmask of resuming clients, to ensure that it does not fall under their list of banned hosts or hostmasks.
+
+
+## Security Considerations
+
+This section notes security-specific considerations software authors will need to take into account while implementing this specification. This section is non-normative.
+
+When servers apply the old client's session information to the new client, ensure that the new client retains their own unique resume token. Clients shouldn't share resume tokens under any circumstances.
+
+Servers must construct resume tokens in a secure way. The [Resume Token](#resume-token) section above lays out a method that should prevent timing attacks. Any implementers creating their own method must protect against timing attacks and other possible attacks against their authentication method.
```

Comment:
I'd add something at the end here like:

>It is recommended that the token provide 128 bits of security strength. In the typical method of generating tokens, this can be achieved by ensuring that the secret "resume key" portion of the token contains 128 bits of entropy from a cryptographically secure pseudorandom source.

Reply:
I think that should be an `"at least 128 bits of strength"` with explanation of why that number was chosen

Reply:
+1 for "at least 128 bits of security strength". Revised comment:

>It is recommended that the token provide at least 128 bits of security strength. This reflects recommendations such as that of NIST [1] that modern systems target a 128-bit security level. In the relevant threat model of online attack by a classical adversary, this can be achieved through the use of a secret with 128 bits of entropy from a cryptographically secure pseudorandom source. With the 'typical method' of generating tokens, it is sufficient to ensure that the 'resume key' portion of the token has this property.
>
>[1] https://csrc.nist.gov/publications/detail/sp/800-57-part-1/rev-4/final

(Not for inclusion in the spec, a more colorful version of the rationale: https://security.stackexchange.com/a/6149)

Reply:
Language is now included, reads well to me and provides some good advice/rationale. Thanks!